src/oncall/scheduler/default.py

```diff
@@ -0,0 +1,320 @@
+        # find people without conflicting events
+        # TODO: finer grain conflict checking
+        user_ids = set(self.get_roster_user_ids(roster_id, cursor))
+        if not user_ids:
+            logger.info('Empty roster, skipping')
+            return None
+        logger.debug('filtering users: %s', user_ids)
+        start = min([e['start'] for e in future_events])
+        for uid in self.get_busy_user_by_event_range(user_ids, team_id, future_events, cursor):
+            user_ids.remove(uid)
+        if not user_ids:
+            logger.info('All users have conflicting events, skipping...')
+            return None
+        new_user_ids = self.find_new_user_in_roster(roster_id, team_id, start, role_id, cursor)
+        available_and_new = new_user_ids & user_ids
+        if available_and_new:
+            logger.info('Picking new and available user from %s', available_and_new)
+            return available_and_new.pop()
+
+        logger.debug('picking user between: %s, team: %s', user_ids, team_id)
+        return self.find_least_active_user_id_by_team(user_ids, team_id, start, role_id, cursor)
+
+    def schedule(self, team, schedules, dbinfo):
+        connection, cursor = dbinfo
+        events = []
+        for schedule in schedules:
+            if schedule['auto_populate_threshold'] <= 0:
+                self.set_last_epoch(schedule['id'], time.time(), cursor)
+                continue
+            logger.info('\t\tschedule: %s', str(schedule['id']))
+            schedule['timezone'] = team['scheduling_timezone']
+            # Calculate events for schedule
+            future_events, last_epoch = self.calculate_future_events(schedule, cursor)
+            for epoch in future_events:
+                # Add (start_time, schedule_id, role_id, roster_id, epoch_events) to events
+                events.append((schedule, epoch))
+            self.set_last_epoch(schedule['id'], last_epoch, cursor)
+
+        # Create events in the db, associating a user to them
+        # Iterate through events in order of start time to properly assign users
+        for schedule, epoch in sorted(events, key=lambda x: min(ev['start'] for ev in x[1])):
+            user_id = self.find_least_active_available_user_id(schedule, epoch, cursor)
+            if not user_id:
+                logger.info('Failed to find available user')
+                continue
+            logger.info('Found user: %s', user_id)
+            self.create_events(team['id'], schedule['id'], user_id, epoch, schedule['role_id'], cursor)
+        connection.commit()
+
+    def populate(self, schedule, start_time, dbinfo):
+        connection, cursor = dbinfo
+        start_dt = datetime.fromtimestamp(start_time, utc)
+        start_epoch = self.epoch_from_datetime(start_dt)
+
+        # Get schedule info
+        role_id = schedule['role_id']
+        team_id = schedule['team_id']
+        first_event_start = min(schedule['events'], key=lambda x: x['start'])['start']
```

Comment:
Would something simpler like this work?

    min(event['start'] for event in schedule['events'])

if not, a oppergetter might be preferred over a lambda